cmd/kubeadm/app/cmd/init.go

```diff
@@ -342,20 +362,41 @@ func newInitData(cmd *cobra.Command, args []string, options *initOptions, out io
 		}
 	}
 
+	var certificateKey []byte
+	if options.uploadCerts {
+		certificateKey, err = cryptoutil.CreateRandBytes(kubeadmconstants.CertificateKeySize)
```

Comment:
> I think that generating the certificateKey should be part of the `app/phases/uploadcerts/uploadcerts.go`

+1 let's have this outside of init.

Reply:
is it ok if I expose the function on `app/phases/uploadcerts/uploadcerts.go` and create the key on init? I need the key to print the join command

Reply:
so i think the key can be stored in initData like the tokens are:
https://github.com/kubernetes/kubernetes/blob/79b08b6222dfdffe2427d172e6d5e29f8bf892b4/cmd/kubeadm/app/cmd/init.go#L497

but the key should be ideally generated by the uploadcerts phase.

from there you can have a similar function to `Tokens()`, but `CertsUploadKey()` that returns it so that it can be fed to printJoinCommand().

at least this is how i see it.

Reply:
Make sense, I'll update soon